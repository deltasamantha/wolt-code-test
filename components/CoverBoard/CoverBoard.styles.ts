import {StyleSheet} from "react-native";

export const styles = StyleSheet.create({
  rootContainer: {
    flex: 1,
    position: "absolute",
    height: "100%",
    width: "100%",
  },
  box: {
    width: 100,
    height: 100,
    borderWidth: 1,
    borderColor: "black",
    position: "absolute",
  },
  box1: {
    backgroundColor: "yellow",
    left: 20,
    top: 0,
  },
  box2: {
    backgroundColor: "red",
    right: 20,
    top: 0,
  },
  box3: {
    backgroundColor: "blue",
    left: 20,
    bottom: 0,
  },
});
