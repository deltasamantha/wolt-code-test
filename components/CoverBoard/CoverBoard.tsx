import React, {useEffect} from "react";
import {View, useWindowDimensions} from "react-native";
import {useSharedValue} from "react-native-reanimated";

import {styles} from "./CoverBoard.styles";
import CoverBoardItem from "../CoverBoardItem/CoverBoardItem";

const CoverBoard: React.FC = () => {
  const translation = useSharedValue<number>(0);
  const {width, height} = useWindowDimensions();

  useEffect(() => {
    translation.value = 1;
  }, []);

  const itemSize = width * 0.8;
  const bottomItemSize = width;

  return (
    <View style={styles.rootContainer}>
      <CoverBoardItem
        source={require("../../assets/images/food-plate-top.png")}
        originX={0}
        originY={-height}
        destinationX={0}
        destinationY={-itemSize / 2}
        style={{width: itemSize, height: itemSize}}
        testID="top"
      />

      <CoverBoardItem
        source={require("../../assets/images/food-plate-middle.png")}
        originX={-width}
        originY={height / 2}
        destinationX={-itemSize / 2}
        destinationY={height / 4}
        style={{width: itemSize, height: itemSize}}
        testID="middle"
      />

      <CoverBoardItem
        source={require("../../assets/images/food-plate-bottom.png")}
        originX={width * 1.4}
        originY={height * 1.5}
        destinationX={width - itemSize * 0.7}
        destinationY={height - itemSize * 0.9}
        style={{width: bottomItemSize, height: bottomItemSize}}
        testID="bottom"
      />
    </View>
  );
};

export default CoverBoard;
