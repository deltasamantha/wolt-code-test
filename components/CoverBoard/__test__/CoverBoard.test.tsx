import {render, cleanup, screen} from "@testing-library/react-native";

import CoverBoard from "../CoverBoard";

afterEach(cleanup);

describe("CoverBoard", () => {
  it("Should render CoverBoard properly", () => {
    const {toJSON} = render(<CoverBoard />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("Should contains Images", () => {
    render(<CoverBoard />);
    const imageItems = screen.getAllByTestId("board-item", {exact: false});
    expect(imageItems).toBeTruthy();
    expect(imageItems.length).toBe(3);
  });
});
