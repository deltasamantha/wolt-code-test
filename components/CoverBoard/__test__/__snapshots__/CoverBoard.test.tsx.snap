// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverBoard Should render CoverBoard properly 1`] = `
<View
  style={
    {
      "flex": 1,
      "height": "100%",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      [
        {
          "position": "absolute",
        },
        {
          "height": 600,
          "width": 600,
        },
        {
          "left": 0,
          "top": -300,
        },
        {
          "transform": [
            {
              "translateX": undefined,
            },
            {
              "translateY": undefined,
            },
          ],
        },
      ]
    }
    testID="top-board-item"
  />
  <Image
    source={1}
    style={
      [
        {
          "position": "absolute",
        },
        {
          "height": 600,
          "width": 600,
        },
        {
          "left": -300,
          "top": 333.5,
        },
        {
          "transform": [
            {
              "translateX": undefined,
            },
            {
              "translateY": undefined,
            },
          ],
        },
      ]
    }
    testID="middle-board-item"
  />
  <Image
    source={1}
    style={
      [
        {
          "position": "absolute",
        },
        {
          "height": 750,
          "width": 750,
        },
        {
          "left": 330,
          "top": 794,
        },
        {
          "transform": [
            {
              "translateX": undefined,
            },
            {
              "translateY": undefined,
            },
          ],
        },
      ]
    }
    testID="bottom-board-item"
  />
</View>
`;
